### PR TITLE
 Update SparseML to support Pytorch 1.10.2

### PR DESCRIPTION
**Pull request by dbogunowicz**
The goal of this PR is to upgrade SparseML to PyTorch 1.10.2 (+ torchvision 0.11.3). 
I have completed following work items:

- **made sure that `torch` tests are reliable.** After the upgrade, some test relating to MFAC, probably tensor inversion, started failing in a stochastic manner. However, with the recent merge from Konstantin, this undesired behaviour has disappeared.
- **made sure that `onnx` test are green.** Refactored `tests/sparseml/onnx/optim/test_analyzer_model.py::test_model_analyzer` and `tests/sparseml/onnx/utils/test_helpers.py::test_extract_node_shape`. Together with @markurtz we have concluded that new PyTorch introduces subtle difference in the structure of the parsed onnx graphs. The short term solution is to introduce tests for both legacy Python and upgraded Python. As long as at least one of them passes, the complete test shall pass. This "hack" will be removed soon with the refactor of the onnx codebase.
- Ran training on ResNet50 and Imaginette using the recipe `integrations/pytorch/recipes/resnet50-pruned_quant.md`. The results are stored in log file `/pytorch_vision_train/tensorboard-logs/resnet50_imagenette__16`. I am not entirely sure how to conclude with high confidence that upgrade did not degrade the training process. Superficially, the result looks okay:
image
image
image

### CODE REVIEW

**Review by bfineran — CHANGES_REQUESTED** [review]
torch 1.10.2 is out and stable, any reason this isn't included in the range?

**Review by bfineran — DISMISSED** [review]

File: src/sparseml/pytorch/base.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
 
 _TORCH_MIN_VERSION = "1.0.0"
-_TORCH_MAX_VERSION = "1.9.100"  # set bug to 100 to support all future 1.9.X  versions
+_TORCH_MAX_VERSION = "1.10.100"  # set bug to 100 to support all future 1.9.X  versions
```
Comment (rahul-tuli) [inline src/sparseml/pytorch/base.py:52]: update comment
```suggestion
_TORCH_MAX_VERSION = "1.10.100"  # set bug to 100 to support all future 1.10.X  versions
```

File: tests/sparseml/onnx/optim/test_analyzer_model.py [hunk 2]
```diff
@@ -154,14 +154,35 @@ def _test_model_analyzer(model_path: str, expected_output: str):
...
+    result = [False] * len(expected_outputs)
+    for i, expected_output in enumerate(expected_outputs):
+        result[i] = _test_model_analyzer(model_path, expected_output)
+    """
+    If we have only one test case, it must must evaluate to True,
+    If we have two test cases, at least one must evaluate to True.
+    In other words, we are happy with test passing for legacy or
+    upgraded PyTorch (worst case scenario).
+    """
+    assert any(result)
```
Comment (rahul-tuli) [inline tests/sparseml/onnx/optim/test_analyzer_model.py:185]: nice

**Review by rahul-tuli — DISMISSED** [review]
The PR is approved barring one small outdated comment, real nice work with the tests. Good work!

File: src/sparseml/pytorch/base.py (hunk 1 above)
Comment (dbogunowicz) [reply src/sparseml/pytorch/base.py:52]: Nice catch! Thank you!

**Review by rahul-tuli — APPROVED** [review]

**Review by KSGulin — APPROVED** [review]
Nice! LGTM

**Pull request merged**